src/pmhn/_trees/_simulate.py

```diff
@@ -31,7 +36,38 @@ def _simulate_tree(
     # TODO(Pawel): This is part of https://github.com/cbg-ethz/pMHN/issues/14
     #   Note that the sampling time is known that our `theta` entries
     #   are log-Theta entries from the paper.
-    raise NotImplementedError
+    print("starting ... \n")
+    theta_size=len(theta)
+    node_time_map={}
+    root=Node("0")
+    node_time_map[root]=0
+    U_current=[root]
+    exit_while=False
+    while len(U_current)!=0:
+        U_next=[]
+        for node in U_current:
+            path=list(node.path)
+            old_mutations=[int(node.name) for node in path]
+            possible_mutations=list(set([i+1 for i in range(theta_size)]).difference(set(old_mutations)))
+            for j in possible_mutations:
+                new_node=Node(str(j),parent=node)
+                l=np.exp(theta[j-1][j-1])
+                for anc in [ancestor for ancestor in node.path if ancestor.parent is not None]:
+                    l*=np.exp(theta[j-1][int(anc.name)-1])
+                waiting_time=node_time_map[node]+rng.exponential(1.0/l)
+                if waiting_time<sampling_time:
+                    node_time_map[new_node]=waiting_time
+                    U_next.append(new_node)
+                    if len(node_time_map)==12:
```

Comment:
I did this because I tried to compare the R implementation with the Python implementation. In the R version the largest_tree_size was 10 by default (excluding the root node), whereas in python max_tree_size stands for the tree size including the root node, meaning max_tree_size = 11. So if the tree size reaches 12 nodes, the function should return, because the tree will be discarded anyway since it doesn't fulfill the size constraints.

Reply:
I see, thanks for explaining! Generally, it's good to [avoid magic numbers](https://www.inf.unibz.it/~calvanese/teaching/04-05-ip/lecture-notes/uni04/node15.html) and in this case the cleanest solution would be just to remove the restriction here. (Filtering trees by size is done by another function. It's a good practice to modularize the code.)